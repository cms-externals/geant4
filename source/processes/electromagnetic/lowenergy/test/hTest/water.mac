#=======================================================
# Macro file "water.mac" for the Water test by hTest.cc
# 10.04.01 V.Ivanchenko
#======================================================= 
/control/verbose           2
/run/verbose               2
/tracking/verbose          0
/event/verbose             0
/hTest/verbose             0
#
/hTest/physics/cutGamma    1.0 mm
/hTest/physics/cutElectron 0.2 mm
/hTest/physics/MaxStep     1.0 mm
/hTest/physics/verbose     0
#/hTest/physics/EMList      LowEnergy
/hTest/physics/EMList      Standard
/hTest/physics/HadronList  LowEnergy
#/hTest/physics/HadronList  none
/hTest/AbsorberMaterial    Water
/hTest/AbsorberThick       1.0 mm
/hTest/AbsorberXY          20.0 cm
/hTest/NumberOfAbsorbers   300
/hTest/WorldMaterial       Vacuum
/hTest/numberAbsToSave     0
/hTest/HistoName           waterstn.paw 
/hTest/NumberOfEvents      1000
/hTest/nFirstEventToDebug -1
/hTest/nLastEventToDebug  -1
/hTest/gun/beamE           175 MeV
/hTest/gun/sigmaE          0.9 MeV
#
/gun/particle              proton
#/gun/particle              e-

