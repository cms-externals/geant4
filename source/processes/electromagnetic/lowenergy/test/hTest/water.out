**********************************************
 Geant4 version 
                                (06-Apr-2001)
             Copyright : Geant4 Collaboration
**********************************************
Detector Construction is defined
HsPhysicsList: <LowEnergy> EMPhysicsList is set
HsPhysicsList: <none> HadronicPhysicsList is set
Physics List is defined
User actions will be initialized
hTestPrimaryGeneratorMessenger: Construct 
hTestPrimaryGeneratorAction is defined
EventAction is defined
TrackingAction is defined
UI interface is started
#=======================================================
# Macro file "water.mac" for the Water test by hTest.cc
# 10.04.01 V.Ivanchenko
#======================================================= 
/run/verbose               2
/step/verbose              0
/tracking/verbose          0
/stepping/verbose          0
/event/verbose             0
/hTest/verbose             0
#
/hTest/physics/cutGamma    1.0 cm
/hTest/physics/cutElectron 1.0 mm
/hTest/physics/MaxStep     1.0 mm
hTestPhysicsList: MaxChargedStep = 1 mm
/hTest/physics/EMList      LowEnergy
#/hTest/physics/EMList      Standard
/hTest/physics/HadronList  LowEnergy
HsPhysicsList: <LowEnergy> HadronicPhysicsList is set
/hTest/AbsorberMaterial    Water
/hTest/AbsorberThick       1.0 mm
/hTest/AbsorberXY          20.0 cm
/hTest/NumberOfAbsorbers   300
/hTest/WorldMaterial       Vacuum
/hTest/numberAbsToSave     0
/hTest/HistoName           water.paw 
/hTest/NumberOfEvents      1000
/hTest/nFirstEventToDebug -1
/hTest/nLastEventToDebug  -1
/hTest/gun/sigmaE          0.9 MeV
hTestPrimaryGeneratorMessenger: O'K 
#
/gun/particle              proton
/gun/energy                175 MeV

Start initialisation
userDetector->Construct() start.

***** Table : Nb of materials = 18 *****

 Material: Beryllium            density:  1.848 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  35.267 cm
   --->  Element: Beryllium    Z =  4.0   N =   9.0   A =   9.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Aluminum            density:  2.700 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.893 cm
   --->  Element: Aluminum    Z = 13.0   N =  27.0   A =  26.98 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Carbon            density:  2.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  21.348 cm
   --->  Element: Carbon    Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Silicon            density:  2.330 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   9.368 cm
   --->  Element: Silicon    Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: liquidArgon            density:  1.390 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  14.065 cm
   --->  Element: liquidArgon    Z = 18.0   N =  40.0   A =  39.95 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Iron            density:  7.870 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.759 cm
   --->  Element: Iron    Z = 26.0   N =  55.9   A =  55.85 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Copper            density:  8.960 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.436 cm
   --->  Element: Copper    Z = 29.0   N =  63.5   A =  63.55 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Gold            density: 19.320 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   3.344 mm
   --->  Element: Gold    Z = 79.0   N = 197.0   A = 196.97 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Lead            density: 11.350 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   5.612 mm
   --->  Element: Lead    Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Water        H_2O   density:  1.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  36.092 cm
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material: Methane        CH_4   density:  0.667 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 696.375 m
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  25.19 %  Abundance  80.00 %
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  74.81 %  Abundance  20.00 %

 Material: Graphite        Graphite   density:  2.265 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  18.834 cm
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: GaAs        GaAs   density:  5.318 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   2.293 cm
   --->  Element: GalliumGa   Z = 31.0   N =  69.7   A =  69.72 g/mole  fractionMass:  48.20 %  Abundance  50.00 %
   --->  Element: ArsenicumAs   Z = 33.0   N =  74.9   A =  74.92 g/mole  fractionMass:  51.80 %  Abundance  50.00 %

 Material: Ethane        C_2H_6   density: 424.100 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.076 m
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  20.16 %  Abundance  75.00 %
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  79.84 %  Abundance  25.00 %

 Material: CsI        CsI   density:  4.530 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.852 cm
   --->  Element: CesiumCs   Z = 55.0   N = 132.9   A = 132.91 g/mole  fractionMass:  51.15 %  Abundance  50.00 %
   --->  Element: IodideI   Z = 53.0   N = 126.9   A = 126.90 g/mole  fractionMass:  48.85 %  Abundance  50.00 %

 Material: Air            density:  1.290 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 285.161 m
   --->  Element: NitrogenN   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %

 Material: Mylar            density:  1.390 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  28.868 cm
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  55.00 %  Abundance  30.30 %
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:   8.33 %  Abundance  54.55 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  36.67 %  Abundance  15.15 %

 Material: Vacuum            density:  0.000 mg/cm3  temperature:   2.73 K  pressure:   0.00 atm  RadLength: 204727576.737 pc
   --->  Element: Vacuum    Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %


The  WORLD   is made of  of Vacuum. The transverse size (XY) of the world is 2e+02 mm
The ABSORBER is made of 300 items of 1 mm of Water. The transverse size (XY) is 2e+02 mm
physicsList->Construct() start.
LowEnergy Electromagnetic PhysicsList is initilized
LowE EM processes for GenericIon
LowE GenericIon
LowE EM processes for He3
LowE He3
LowE EM processes for IonAr40
LowE IonAr40
LowE EM processes for IonC12
LowE IonC12
LowE EM processes for IonFe56
LowE IonFe56
LowE EM processes for alpha
LowE alpha
LowE EM processes for anti_kaon0
LowE EM processes for anti_lambda
LowE EM processes for anti_neutron
LowE EM processes for anti_nu_e
LowE EM processes for anti_nu_mu
LowE EM processes for anti_omega-
LowE EM processes for anti_proton
LowE anti_proton
LowE EM processes for anti_sigma+
LowE EM processes for anti_sigma-
LowE EM processes for anti_sigma0
LowE EM processes for anti_xi-
LowE EM processes for anti_xi0
LowE EM processes for chargedgeantino
LowE EM processes for deuteron
LowE deuteron
LowE EM processes for e+
LowE e+
LowE EM processes for e-
LowE e-
>>>>>>>>> New G4LowEnergyIonisation 0x8932e40
LowE EM processes for eta
LowE EM processes for eta_prime
LowE EM processes for gamma
LowE gamma
LowE EM processes for geantino
LowE EM processes for kaon+
LowE kaon+
LowE EM processes for kaon-
LowE kaon-
LowE EM processes for kaon0
LowE EM processes for kaon0L
LowE EM processes for kaon0S
LowE EM processes for lambda
LowE EM processes for mu+
LowE mu+
MuonMinusCaptureAtRest is created 
LowE EM processes for mu-
LowE mu-
MuonMinusCaptureAtRest is created 
LowE EM processes for neutron
LowE EM processes for nu_e
LowE EM processes for nu_mu
LowE EM processes for omega-
LowE EM processes for pi+
LowE pi+
LowE EM processes for pi-
LowE pi-
LowE EM processes for pi0
LowE EM processes for proton
LowE proton
LowE EM processes for sigma+
LowE sigma+
LowE EM processes for sigma-
LowE sigma-
LowE EM processes for sigma0
LowE EM processes for triton
LowE triton
LowE EM processes for xi-
LowE EM processes for xi0
LowEnergy Hadronic PhysicsList is initilized
physicsList->setCut() start.
hTestPhysicsList: Set cuts for all particles 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV to 1e+02 TeV in 100 bins. 
*** LEion using Tcut =0.47
*** LEion using Tcut =0.6
*** LEion using Tcut =0.52
*** LEion using Tcut =0.55
*** LEion using Tcut =0.34
*** LEion using Tcut =1.3
*** LEion using Tcut =1.4
*** LEion using Tcut =2.3
*** LEion using Tcut =1.4
*** LEion using Tcut =0.35
*** LEion using Tcut =0.00012
*** LEion using Tcut =0.57
*** LEion using Tcut =0.87
*** LEion using Tcut =0.22
*** LEion using Tcut =0.69
*** LEion using Tcut =0.00028
*** LEion using Tcut =0.43
*** LEion using Tcut =1e-05

LowEnergyIoni:  Low energy ionisation code with implementation of the continuous dE/dx part.
 At present it can be used for electrons  in the energy range [250eV,100GeV]
 the process must work with G4LowEnergyBremsstrahlung
*** LE Bremsstrahlung using Gamma Tcut = 0.0038 for material Beryllium
*** LE Bremsstrahlung using Gamma Tcut = 0.02 for material Aluminum
*** LE Bremsstrahlung using Gamma Tcut = 0.0075 for material Carbon
*** LE Bremsstrahlung using Gamma Tcut = 0.021 for material Silicon
*** LE Bremsstrahlung using Gamma Tcut = 0.019 for material liquidArgon
*** LE Bremsstrahlung using Gamma Tcut = 0.061 for material Iron
*** LE Bremsstrahlung using Gamma Tcut = 0.07 for material Copper
*** LE Bremsstrahlung using Gamma Tcut = 0.38 for material Gold
*** LE Bremsstrahlung using Gamma Tcut = 0.28 for material Lead
*** LE Bremsstrahlung using Gamma Tcut = 0.0077 for material Water
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Methane
*** LE Bremsstrahlung using Gamma Tcut = 0.0079 for material Graphite
*** LE Bremsstrahlung using Gamma Tcut = 0.064 for material GaAs
*** LE Bremsstrahlung using Gamma Tcut = 0.0039 for material Ethane
*** LE Bremsstrahlung using Gamma Tcut = 0.1 for material CsI
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Air
*** LE Bremsstrahlung using Gamma Tcut = 0.0075 for material Mylar
*** LE Bremsstrahlung using Gamma Tcut = 1e-05 for material Vacuum

LowEnBrem:  Total cross sections from EEDL database,Gamma energy sampled from a parametrised formula.Implementation of the continuous dE/dx part.
 At present it can be used for electrons  in the energy range [250eV,100GeV]
 the process must work with G4LowEnergyIonisation

annihil:  Total cross section from Heilter formula (annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV to 1e+02 TeV in 100 bins. 

MuIoni:  knock-on electron cross sections .
          Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
      PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

MuBrems:  theoretical cross section 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

MuPairProd:  theoretical cross sections 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV to 1e+02 TeV in 100 bins. 

hLowEIoni:    Knock-on electron cross sections . 
        Good description above the mean excitation energy.
        Delta ray energy sampled from  differential Xsection.
        PhysicsTables from 10 eV  to 1e+02 TeV  in 200 bins.
        Electronic stopping power model is  ICRU_R49p
        from 1 keV  to 2 MeV .

        Parametrisation model for antiprotons is  ICRU_R49p
        from 50 keV  to 2 MeV .
        Parametrization of the Barkas effect is switched on.
        Nuclear stopping power model is ICRU_R49
           material       min.delta energy(keV) 

Vacuum          0.021
============= The cut Energy ==============================
                      gamma            e-           
Cut in range                  1 cm             1 mm
Cut in energy
 Beryllium                         3.84 keV           467 keV
   Aluminum                         20.4 keV           597 keV
   Carbon                         7.51 keV           522 keV
   Silicon                         20.7 keV           546 keV
   liquidArgon                         19.2 keV           345 keV
   Iron                         60.6 keV          1.29 MeV
   Copper                         70.1 keV          1.41 MeV
   Gold                          377 keV          2.28 MeV
   Lead                          284 keV          1.38 MeV
   Water                         7.72 keV           353 keV
   Methane                           10 eV           119 eV
   Graphite                         7.92 keV           571 keV
   GaAs                         63.9 keV           873 keV
   Ethane                         3.87 keV           218 keV
   CsI                          103 keV           690 keV
   Air                           10 eV           276 eV
   Mylar                         7.49 keV           431 keV
   Vacuum                           10 eV            10 eV
===================================================
Start event loop for 1000 events
hTestHisto: Histograms will be saved to the file <water.paw>
Start closing geometry.
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 1000
  User=23s Real=23s Sys=0.21s
hTestHisto: End of run actions are started
========================================================
Range(mm)= 220.5; Stragling(mm)= 62.36 +- 2
========================================================
Histograms and Ntuples are saved
runManager will be deleted
G4 kernel has come to Quit state.
Deletion of G4 kernel class start.
UserDetectorConstruction deleted.
UserPhysicsList deleted.
UserPrimaryGenerator deleted.
G4SDManager deleted.
EventManager deleted.
UImanager deleted.
StateManager deleted.
RunManager is deleting.
runManager is deleted
